android/sdl_android/src/androidTest/java/com/smartdevicelink/managers/lockscreen/LockScreenDeviceIconManagerTests.java

```diff
@@ -0,0 +1,242 @@
+        Mockito.when(context.getSharedPreferences(anyString(), anyInt())).thenReturn(sharedPrefs);
+        Mockito.when(sharedPrefs.edit()).thenReturn(sharedPrefsEditor);
+        Mockito.when(sharedPrefsEditor.remove(anyString())).thenReturn(sharedPrefsEditor);
+        Mockito.when(sharedPrefsEditor.commit()).thenReturn(true);
+        Mockito.when(sharedPrefs.getString(anyString(), (String) isNull())).thenReturn(buildJSONAsString(15, ""));
+
+        try {
+            tempFolder.create();
+            Mockito.when(context.getCacheDir()).thenReturn(tempFolder.newFolder());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+
+        lockScreenDeviceIconManager = new LockScreenDeviceIconManager(context);
+        Bitmap cachedIcon = lockScreenDeviceIconManager.getFileFromCache(ICON_URL);
+        assertNull(cachedIcon);
+    }
+
+    public void testGetFileFromCacheShouldReturnBitmapIfIconFoundInCache() {
+        final SharedPreferences sharedPrefs = Mockito.mock(SharedPreferences.class);
+        final SharedPreferences.Editor sharedPrefsEditor = Mockito.mock(SharedPreferences.Editor.class);
+        final Context context = Mockito.mock(Context.class);
+        Mockito.when(context.getSharedPreferences(anyString(), anyInt())).thenReturn(sharedPrefs);
+        Mockito.when(sharedPrefs.edit()).thenReturn(sharedPrefsEditor);
+        Mockito.when(sharedPrefsEditor.remove(anyString())).thenReturn(sharedPrefsEditor);
+        Mockito.when(sharedPrefsEditor.commit()).thenReturn(true);
+        Bitmap deviceLogo = null;
+
+        try {
+            tempFolder.create();
+            File newFolder = tempFolder.newFolder();
+            Mockito.when(context.getCacheDir()).thenReturn(newFolder);
+            deviceLogo = AndroidTools.downloadImage(ICON_URL);
+            Mockito.when(sharedPrefs.getString(anyString(), (String) isNull())).thenReturn(buildJSONAsString(15, newFolder.getPath() + "/sdl/lock_screen_icon/" + getMD5HashFromIconUrl(ICON_URL)));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        lockScreenDeviceIconManager = new LockScreenDeviceIconManager(context);
+        lockScreenDeviceIconManager.saveFileToCache(deviceLogo, ICON_URL);
+        Bitmap cachedIcon = lockScreenDeviceIconManager.getFileFromCache(ICON_URL);
+        assertNotNull(cachedIcon);
+    }
+
+    private String buildJSONAsString(long DaysOld, String cahceIconUrl) {
+        JSONObject jsonObject = new JSONObject();
+        try {
+            jsonObject.put(STORED_URL, cahceIconUrl);
+            long timeDifferenceInMilliSeconds = DaysOld * 1000 * 60 * 60 * 24;
+            jsonObject.put(LAST_UPDATED_TIME, System.currentTimeMillis() - timeDifferenceInMilliSeconds);
+            return jsonObject.toString();
+        } catch (JSONException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    private String getMD5HashFromIconUrl(String iconUrl) {
```

Comment:
Would it be possible to reuse the method in `LockScreenDeviceIconManager()` by making it package-private?

Reply:
The way it is implemented I need to define the mock (using the md5 hash method) before the class in instantiated, if the class is instantiated before the mock is defined it breaks the test.